numpy/core/numerictypes.py

```diff
@@ -421,7 +423,7 @@ def issubdtype(arg1, arg2):
 
 
 # This dictionary allows look up based on any alias for an array data-type
-class _typedict(dict):
+class _typedict(dict[Any, Any]):
```

Comment:
Honestly I only tried this in a python shell but I am unsure if this might have undesired side effects.

```python
In [1]: from typing import Any

In [2]: class tdict(dict[Any, Any]):
   ...:     ...
   ...:

In [3]: tdict()
Out[3]: {}
```

Reply:
Can you remove any python file changes?  We don't have typing in the code, only stubs.  So I don't understand why it should be necessary or even useful.  Or is there some good reason for it?

Reply:
Sure, i dont have any good reason for it